specs-go/v1/manifest.go

```diff
@@ -36,3 +36,10 @@ type Manifest struct {
 	// Annotations contains arbitrary metadata for the image manifest.
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
+
+// ScratchConfigDescriptor is a SCRATCH descriptor for Manifest's `config` when that manifest does not have a config to provide.
+var ScratchConfigDescriptor = imagespec.Descriptor{
```

Comment:
```suggestion
var ScratchConfigDescriptor = Descriptor{
```

👀

(Is it worth making this a function to ensure it stays immutable, or do we trust consumers of it to not screw it up? Probably the latter?)

Reply:
... maybe?
but then I would also have the const of the digest as string for easier comparion, rather than requiring a function callZ? maybe?

Reply:
Yeah, seems like a lot of overhead just to protect against clients (accidentally) messing with the variable.  Go really ought to have something better here :cry:

Reply:
Should we just define the data and digest as constants due to these limitations? The main ask here is to codify it in the spec which is done in - 
https://github.com/opencontainers/image-spec/pull/1023/files#diff-7333951212fd75b50f5be1023afc9d5d41f0533adc240407c43d4bfb42c6019aR131-R138

Reply:
We could also define this as a function that takes a media type string and returns a new descriptor? Takes care of the mutability concern and is a straightforward interface for setting the artifact type.